class/practice_4.1/pizza.cpp

```diff
@@ -0,0 +1,68 @@
+#include "pizza.hpp"
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <map>
+
+Pizza::Pizza() : total_price(0.0) {}
+
+void Pizza::addIngredient(const Ingredient& ingredient) {
+    ingredients.push_back(ingredient);
+    total_price += ingredient.price;
+}
+
+double Pizza::getTotalPrice() const {
+    return total_price;
+}
+
+const std::vector<Ingredient>& Pizza::getIngredients() const {
+    return ingredients;
+}
+
+PizzaBuilder::PizzaBuilder(const std::string& ingredientsFile) : ingredientsFile(ingredientsFile) {}
+
+Pizza PizzaBuilder::buildPizza() {
+    Pizza pizza;
+    loadIngredients();
+
+    std::cout << "Available ingredients:\n";
```

Comment:
interaction with user should not be a part of this class as it violates single responsibility principle. builder should provide an API to build pizza step by step
Also those prompts make unit testiing of this class hard. If prompts will b eon higher level then you can easy cover PiizzaBuilder api with unit tests